guardrails/llm_providers.py

```diff
@@ -381,15 +564,45 @@ async def invoke_llm(
             *args,
             **kwargs,
         )
-        return LLMResponse(
-            output=openai_response["choices"][0]["text"],  # type: ignore
-            prompt_token_count=openai_response["usage"][  # type: ignore
-                "prompt_tokens"
-            ],
-            response_token_count=openai_response["usage"][  # type: ignore
-                "completion_tokens"
-            ],
-        )
+
```

Comment:
yeah same here. In general, the forking is alright, but I think the issue is just that the function becomes big and hard to read. Instead, at the conditional forks, let's split up the execution into functions, set the vars that are needed adn return afterwards.